backends/imgui_impl_d3d9.cpp

```diff
@@ -0,0 +1,490 @@
+    SafeAddRef(g.d3d9Device);
+    
+    return true;
+}
+
+void ImGui_ImplDX9_Shutdown()
+{
+    ImGui_ImplDX9_InvalidateDeviceObjects();
+    g.reset();
+}
+
+static bool ImGui_ImplDX9_CreateFontsTexture()
+{
+    // Build texture atlas
+    ImGuiIO& io = ImGui::GetIO();
+    unsigned char* pixels;
+    int width, height, bytes_per_pixel;
+    io.Fonts->GetTexDataAsRGBA32(&pixels, &width, &height, &bytes_per_pixel);
+    
+    // Create a texture without D3DUSAGE_DYNAMIC (static texture)
+    // https://github.com/Microsoft/DirectXTex
+    
+    // 1. Pick a texture pixel format
+#ifndef IMGUI_USE_BGRA_PACKED_COLOR
+    D3DFORMAT texture_format = D3DFMT_A8B8G8R8; // RGBA32, be aware that this format is not widely supported on Windows platfrom
+#else
+    D3DFORMAT texture_format = D3DFMT_A8R8G8B8; // BGRA32
+#endif
+    
+    // 2. Create staging texture and write pixel data
+    IDirect3DTexture9* texture_staging = NULL;
+    struct com_helper
+    {
+        // This is a helper to manage COM object
+        // We can also using Microsoft::WRL::ComPtr (wrl.h) or CComPtr (ATL)
+        IDirect3DTexture9*& ptr;
+        com_helper(IDirect3DTexture9*& tex) : ptr(tex) {}
+        ~com_helper() { if (ptr) { ptr->Release(); ptr = NULL; } }
+    } texture_staging_ref(texture_staging);
+    if (D3D_OK != g.d3d9Device->CreateTexture(width, height, 1, D3DUSAGE_DYNAMIC, texture_format, D3DPOOL_SYSTEMMEM, &texture_staging, NULL))
+        return false;
+    D3DLOCKED_RECT tex_locked_rect;
+    if (D3D_OK != texture_staging->LockRect(0, &tex_locked_rect, NULL, 0))
+        return false;
+    for (int y = 0; y < height; y++)
+        memcpy((unsigned char*)tex_locked_rect.pBits + tex_locked_rect.Pitch * y, pixels + (width * bytes_per_pixel) * y, (width * bytes_per_pixel));
+    if (D3D_OK != texture_staging->UnlockRect(0))
+        return false;
+    
+    // 3. Upload to graphic card memory
+    SafeRelease(g.d3d9Texture2D);
+    if (D3D_OK != g.d3d9Device->CreateTexture(width, height, 1, 0, texture_format, D3DPOOL_DEFAULT, &g.d3d9Texture2D, NULL))
+        return false;
+    if (D3D_OK != g.d3d9Device->UpdateTexture(texture_staging, g.d3d9Texture2D))
+        return false;
+    
+    // Store our identifier
+    io.Fonts->SetTexID((ImTextureID)g.d3d9Texture2D);
+    
+    return true;
+}
+
+bool ImGui_ImplDX9_CreateDeviceObjects()
+{
+    if (!g.d3d9Device)
+        return false;
+    
+    // Create input layout
+    D3DVERTEXELEMENT9 layout[4] = {
+        { 0,  0, D3DDECLTYPE_FLOAT2  , D3DDECLMETHOD_DEFAULT, D3DDECLUSAGE_POSITION, 0 },
+        { 0,  8, D3DDECLTYPE_FLOAT2  , D3DDECLMETHOD_DEFAULT, D3DDECLUSAGE_TEXCOORD, 0 },
+        { 0, 16, D3DDECLTYPE_D3DCOLOR, D3DDECLMETHOD_DEFAULT, D3DDECLUSAGE_COLOR   , 0 },
+        D3DDECL_END(),
+    };
+    SafeRelease(g.d3d9InputLayout);
+    if (D3D_OK != g.d3d9Device->CreateVertexDeclaration(layout, &g.d3d9InputLayout))
+        return false;
+    
+    // Create vertex shader (vs_3_0)
```

Comment:
You can use vs_1_1 instead of vs_3_0, also ps_2_0 instead of ps_3_0.
There are not using Shader Model 3.0 Features here.

Also you can see my assembly shader code that more readable and editable without shader compiling.
https://github.com/metarutaiga/xxGraphic/blob/master/xxGraphicD3DAsm.cpp